dartagnan/src/main/java/com/dat3m/dartagnan/wmm/relation/binary/RelComposition.java

```diff
@@ -78,6 +84,46 @@ public TupleSet getMaxTupleSetRecursive(){
         return getMaxTupleSet();
     }
 
+	@Override
+	public boolean disable(TupleSet t) {
+		super.disable(t);
+		if(t.isEmpty())
+			return false;
+		BranchEquivalence eq = task.getBranchEquivalence();
+		TupleSet t1 = new TupleSet();
+		TupleSet t2 = new TupleSet();
+		for(Tuple tuple : t) {
+			Event x = tuple.getFirst();
+			Event z = tuple.getSecond();
+			if(z.cfImpliesExec())
+				r1.getMinTupleSet().getByFirst(x).stream()
+				.map(Tuple::getSecond)
+				.filter(y -> eq.isImplied(y, z))
+				.map(y -> new Tuple(y, z))
+				.filter(r2.getMaxTupleSet()::contains)
+				.forEach(t2::add);
```

Comment:
I think this code is backwards. To disable `r2(y, z)` you need to argue that its existence would cause a violation.
That happens if either `y` or `z` imply `x` and `r1(x, y)` is a must-edge.
The argument is then: Assume `r2(y, z)` was true, then `x` would be in the execution and hence we also have `r1(x,y)` because it is a must-edge. But then we get `r(x, z)` which should be false (contradiction).

Reply:
Good point, this has to be fixed asap.
```suggestion
			if(x.cfImpliesExec())
				r1.getMinTupleSet().getByFirst(x).stream()
				.map(Tuple::getSecond)
				.filter(y -> eq.isImplied(z,x) || eq.isImplied(y, x))
				.map(y -> new Tuple(y, z))
				.filter(r2.getMaxTupleSet()::contains)
				.forEach(t2::add);
```